generators/entity-client/templates/react/src/main/webapp/app/entities/_entity.tsx

```diff
@@ -45,13 +54,27 @@ export interface I<%= entityReactName %>Props {
   <%_ } _%>
   match: any;
 }
+<%_ if (searchEngine === 'elasticsearch') { _%>
 
-export class <%= entityReactName %> extends React.Component<I<%= entityReactName %>Props, undefined> {
+export interface I<%= entityReactName %>State {
+  search: string;
+}
+<%_ } _%>
+
+export class <%= entityReactName %> extends React.Component<I<%= entityReactName %>Props, <% if (searchEngine === 'elasticsearch') { %>I<%= entityReactName %>State> {
 
   constructor(props) {
     super(props);
+    this.state = {
+      search: ''
+    };
   }
+<% } else { %>undefined> {
```

Comment:
I guess you misunderstood me, for a stateless component constructor and the undefined in generic is not needed. I'll update this

Reply:
ah yes sorry you're right, you fix this ?